daemon/mgr/container_utils.go

```diff
@@ -379,3 +384,77 @@ func amendContainerSettings(config *types.ContainerConfig, hostConfig *types.Hos
 		r.MemorySwap = 2 * r.Memory
 	}
 }
+
+// validateNvidiaConfig
+func validateNvidiaConfig(hostConfig *types.HostConfig) error {
+	r := &hostConfig.Resources
+	if r.NvidiaConfig == nil {
+		return nil
+	}
+
+	if err := vlidateNvidiaDriver(r); err != nil {
+		return err
+	}
+
+	if err := vlidateNvidiaDevice(hostConfig); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func stringInSlice(a string, list []string) bool {
```

Comment:
You do not need to add this function, this has been covered in `utils.StringInSlice`.